### ISSUE TITLE
Cannot read property 'test' of null

### ISSUE THREAD

**Issue by giampaolo44** [cypress-io/cypress#17929]
### Current behavior

Possibly a variant of #5789 @jennifer-shehane 

A few days ago my test suite stopped working. It happened after the upgrade to Wp5 and WDS 4 (mentioned it in the Gitter chat and in [this SO question](https://stackoverflow.com/questions/68914580/cypress-complains-on-a-webpack-build-of-pdfjs-dist-which-works-in-development)). In the attempt to debug I reverted to the previous setup:
```
    "webpack": "^4.43.0",
    "webpack-bundle-tracker": "^1.0.0-alpha.1",
    "webpack-cli": "^3.3.12",
    "webpack-dev-server": "^3.11.0",
```
The error morphed into this one:
```
An uncaught error was detected outside of a testfailed
No commands were issued in this test.
TypeError
The following error originated from your test code, not from Cypress.

  > Cannot read property 'test' of null

When Cypress detects uncaught errors originating from your test code it will automatically fail the current test.

Cypress could not associate this error to any specific test.

We dynamically generated a new test to display this failure.
src/js/common/settings.js:13:67
  11 | // --> noUIlogin, and
  12 | // --> command "disableLogging"
> 13 | export const globalDebug = () => !document.querySelector("#home").test === "testing"
     |                                                                   ^
  14 | 
  15 | 
  16 | // list here the thisFunctionIs parameters 
View stack trace
 Print to console
    at globalDebug (webpack:///./src/js/common/settings.js:13:67)
    at thisFunctionIs (webpack:///./src/js/common/logDebug.js:164:8)
    at Object../src/js/show/index-show.js (webpack:///./src/js/show/index-show.js:106:25)
    at __webpack_require__ (webpack:///webpack/bootstrap:19:1)
    at Object../src/js/select/preview/renderPreview.js (webpack:///./src/js/select/preview/renderPreview.js:21:1)
    at __webpack_require__ (webpack:///webpack/bootstrap:19:1)
    at Object../src/js/select/views/rectViews.js (webpack:///./src/js/select/views/rectViews.js:9:1)
    at __webpack_require__ (webpack:///webpack/bootstrap:19:1)
    at Object../src/js/select/utils/DOMutils/cancelRect.js (webpack:///./src/js/select/utils/DOMutils/cancelRect.js:2:1)
    at __webpack_require__ (webpack:///webpack/bootstrap:19:1)
From previous event:
    at runScriptsFromUrls (http://localhost:46041/__cypress/runner/cypress_runner.js:160060:98)
    at Object.runScripts (http://localhost:46041/__cypress/runner/cypress_runner.js:160074:11)
    at $Cypress.onSpecWindow (http://localhost:46041/__cypress/runner/cypress_runner.js:149502:19)
```
Console version:
```
cypress_runner.js:173240 TypeError: The following error originated from your test code, not from Cypress.

  > Cannot read property 'test' of null

When Cypress detects uncaught errors originating from your test code it will automatically fail the current test.

Cypress could not associate this error to any specific test.

We dynamically generated a new test to display this failure.
    at globalDebug (http://localhost:46041/__cypress/tests?p=cypress/support/index.js:1438:59)
    at thisFunctionIs (http://localhost:46041/__cypress/tests?p=cypress/support/index.js:1365:34)
    at Object../src/js/show/index-show.js (http://localhost:46041/__cypress/tests?p=cypress/support/index.js:8957:54)
    at __webpack_require__ (http://localhost:46041/__cypress/tests?p=cypress/support/index.js:20:30)
    at Object../src/js/select/preview/renderPreview.js (http://localhost:46041/__cypress/tests?p=cypress/support/index.js:4288:18)
    at __webpack_require__ (http://localhost:46041/__cypress/tests?p=cypress/support/index.js:20:30)
    at Object../src/js/select/views/rectViews.js (http://localhost:46041/__cypress/tests?p=cypress/support/index.js:5925:22)
    at __webpack_require__ (http://localhost:46041/__cypress/tests?p=cypress/support/index.js:20:30)
    at Object../src/js/select/utils/DOMutils/cancelRect.js (http://localhost:46041/__cypress/tests?p=cypress/support/index.js:4530:18)
    at __webpack_require__ (http://localhost:46041/__cypress/tests?p=cypress/support/index.js:20:30)
From previous event:
    at runScriptsFromUrls (http://localhost:46041/__cypress/runner/cypress_runner.js:160060:97)
    at Object.runScripts (http://localhost:46041/__cypress/runner/cypress_runner.js:160074:10)
    at $Cypress.onSpecWindow (http://localhost:46041/__cypress/runner/cypress_runner.js:149502:18)
logError @ cypress_runner.js:173240
```
my Cypress.json:
```
{
  "ignoreTestFiles": [
    "**/__snapshots__/*",
    "**/__image_snapshots__/*"
  ],
  "viewportWidth": 1980,
  "viewportHeight": 1664,
  "pluginsFile": false,
  "modifyObstructiveCode": false,
  "experimentalFetchPolyfill": true
}
```
I tried the "experimentalFetchPolyfill" suggestion mentioned in #5789 to no avail.

The situation does not change when upgrading to 8.3.1:
Left panel messages:
```
An uncaught error was detected outside of a testfailed
No commands were issued in this test.
TypeError
The following error originated from your test code, not from Cypress.

  > Cannot read property 'test' of null

When Cypress detects uncaught errors originating from your test code it will automatically fail the current test.

Cypress could not associate this error to any specific test.

We dynamically generated a new test to display this failure.
src/js/common/settings.js:13:67
  11 | // --> noUIlogin, and
  12 | // --> command "disableLogging"
> 13 | export const globalDebug = () => !document.querySelector("#home").test === "testing"
     |                                                                   ^
  14 | 
  15 | 
  16 | // list here the thisFunctionIs parameters 
View stack trace
 Print to console
    at globalDebug (webpack:///./src/js/common/settings.js:13:67)
    at thisFunctionIs (webpack:///./src/js/common/logDebug.js:164:8)
    at Object../src/js/show/index-show.js (webpack:///./src/js/show/index-show.js:106:25)
    at __webpack_require__ (webpack:///webpack/bootstrap:19:1)
    at Object../src/js/select/preview/renderPreview.js (webpack:///./src/js/select/preview/renderPreview.js:21:1)
    at __webpack_require__ (webpack:///webpack/bootstrap:19:1)
    at Object../src/js/select/views/rectViews.js (webpack:///./src/js/select/views/rectViews.js:9:1)
    at __webpack_require__ (webpack:///webpack/bootstrap:19:1)
    at Object../src/js/select/utils/DOMutils/cancelRect.js (webpack:///./src/js/select/utils/DOMutils/cancelRect.js:2:1)
    at __webpack_require__ (webpack:///webpack/bootstrap:19:1)
From previous event:
    at runScriptsFromUrls (http://localhost:39313/__cypress/runner/cypress_runner.js:164985:98)
    at Object.runScripts (http://localhost:39313/__cypress/runner/cypress_runner.js:164999:11)
    at $Cypress.onSpecWindow (http://localhost:39313/__cypress/runner/cypress_runner.js:153977:19)
```
Console:
```
cypress_runner.js:178630 TypeError: The following error originated from your test code, not from Cypress.

  > Cannot read property 'test' of null

When Cypress detects uncaught errors originating from your test code it will automatically fail the current test.

Cypress could not associate this error to any specific test.

We dynamically generated a new test to display this failure.
    at globalDebug (http://localhost:39313/__cypress/tests?p=cypress/support/index.js:1438:59)
    at thisFunctionIs (http://localhost:39313/__cypress/tests?p=cypress/support/index.js:1365:34)
    at Object../src/js/show/index-show.js (http://localhost:39313/__cypress/tests?p=cypress/support/index.js:8957:54)
    at __webpack_require__ (http://localhost:39313/__cypress/tests?p=cypress/support/index.js:20:30)
    at Object../src/js/select/preview/renderPreview.js (http://localhost:39313/__cypress/tests?p=cypress/support/index.js:4288:18)
    at __webpack_require__ (http://localhost:39313/__cypress/tests?p=cypress/support/index.js:20:30)
    at Object../src/js/select/views/rectViews.js (http://localhost:39313/__cypress/tests?p=cypress/support/index.js:5925:22)
    at __webpack_require__ (http://localhost:39313/__cypress/tests?p=cypress/support/index.js:20:30)
    at Object../src/js/select/utils/DOMutils/cancelRect.js (http://localhost:39313/__cypress/tests?p=cypress/support/index.js:4530:18)
    at __webpack_require__ (http://localhost:39313/__cypress/tests?p=cypress/support/index.js:20:30)
From previous event:
    at runScriptsFromUrls (http://localhost:39313/__cypress/runner/cypress_runner.js:164985:97)
    at Object.runScripts (http://localhost:39313/__cypress/runner/cypress_runner.js:164999:10)
    at $Cypress.onSpecWindow (http://localhost:39313/__cypress/runner/cypress_runner.js:153977:18)
logError @ cypress_runner.js:178630
```
And terminal output, for good measure:
```
(base) [giampaolo@giampaolo KJ-JS]$ npx cypress open
[39296:0830/003039.073677:WARNING:vaapi_wrapper.cc(588)] VAAPI video acceleration not available for swiftshader
[39296:0830/003039.073868:ERROR:gpu_init.cc(440)] Passthrough is not supported, GL is swiftshader
GET /__/ 200 11.720 ms - -
GET /__cypress/runner/cypress_runner.css 200 11.475 ms - -
GET /__cypress/runner/cypress_runner.js 200 16.496 ms - -
GET /chrome-variations/seed?osname=linux&channel=stable&milestone=92 304 584.579 ms - -
POST /ListAccounts?gpsia=1&source=ChromiumBrowser&json=standard 200 584.653 ms - -
GET /__cypress/static/favicon.ico 200 3.530 ms - -
GET /__cypress/iframes/integration/begin/KJi_login.spec.js 200 4.181 ms - 940
GET /__cypress/runner/fonts/fa-solid-900.woff2 200 15.065 ms - 76120
GET /__cypress/runner/fonts/open-sans-latin-400-normal.woff2 200 2.723 ms - 14380
GET /__cypress/runner/fonts/mulish-latin-500-normal.woff2 200 23.817 ms - 31176
GET /__cypress/runner/fonts/mulish-latin-400-normal.woff2 200 28.695 ms - 31176
GET /v1/pages/ChRDaHJvbWUvOTIuMC40NTE1LjE1ORIQCcmLdcaTM2VsEgUNkWGVThIXCQPNs0E5ZBKtEgUNL4g-ghIFDZFhlU4YqqnKAQ==?alt=proto 200 580.460 ms - -
GET /__cypress/tests?p=cypress/integration/begin/KJi_login.spec.js 200 1051.301 ms - -
GET /__cypress/tests?p=cypress/support/index.js 200 3196.702 ms - -
GET /__cypress/runner/fonts/fa-regular-400.woff2 200 1.722 ms - 13600
GET /__cypress/runner/fonts/mulish-latin-700-normal.woff2 200 1.888 ms - 31176
HEAD / - - ms - -
HEAD / - - ms - -
HEAD / - - ms - -
GET /__cypress/runner/cypress_runner.css 304 0.975 ms - -
GET /__cypress/runner/popper.js.map 404 0.790 ms - 136
```

### Desired behavior

The test suite should run. No errors are shown in the application nor in webpack compilation.
Webpack terminal output:
```
Version: webpack 4.46.0
Time: 80ms
Built at: 2021-08-29 23:37:00
    Asset     Size  Chunks             Chunk Names
select.js  870 KiB  select  [emitted]  select
  show.js  851 KiB    show  [emitted]  show
Entrypoint select = select.js
Entrypoint show = show.js
[./src/js/show/utils/base-show.js] 414 bytes {select} {show} [built]
    + 125 hidden modules
ℹ ｢wdm｣: Compiled successfully.
```

### Test code to reproduce

Unfortunately I am not in condition to provide a failing test. Part of it is that I am not experienced enough to dig what's going wrong in my >100 modules, and can't post them either.
(Forked the `cypress-test-tiny` and it works fine)

### Cypress Version

tested in 7.7.0 and 8.3.1

### Other

As you might infer from the messages: the tests do no start, so the error seems odd. How could the test suite check if a DOM element exists if it does not visit it?

Anyway: I am available for more tests or details.

**Comment by jennifer-shehane** [comment]
Where if this code coming from? Seems the `document.querySelector("#home")` in this situation is null, have you inspected where this comes from?

```
export const globalDebug = () => !document.querySelector("#home").test === "testing"
```

**Comment by giampaolo44** [comment]
Hi @jennifer-shehane, yes: it is the topmost element in the main page of the application. It is there as soon as the app is loaded.
This function has been working and testing fine in Cy for at least a couple of months.
(Background is: Cy changes that property into "testing" at tests start. That way console logging is prevented, greatly improving test speed. I.e. when tests start globalDebug is returned as false.)

**Comment by giampaolo44** [comment]
I sorted this issue out, finally. It appears Cypress does not like certain circular references. They came out converting some text elements into JS variables. Those modules were imported from two different entry points in my Webpack bundle, and this created the whole issue.

The main trouble is: code would work almost perfectly in the dev environment. How comes Cy could not run it inside its own? I could not figure it out, most likely out my abilities.

I could appreciated the heads up and the fact that Cypress is being more demanding than in normal execution/development (sort of a "use strict" thing), but I would highly recommend better logging and possibly some work done on the stack trace? 

I am saying this because the problem came out of the blue, and for days my tests were not running at all! It was not easy to single the issues out and untangling the problem progressively until I finally got it right. 

Felt a bit ignored too.
